### ISSUE TITLE
Report Loopholes Here

### ISSUE THREAD

**Issue by dr-hextanium** [dev-launchers/minecraft__byte-economy#2]
This issue is for reporting any potential loopholes. 
Users who report loopholes in the system will be rewarded, and appreciated.

**Comment by dr-hextanium** [comment]
<h1>NOTE: The following is an example loophole, along with the format.</h1>
<h2>Mob Farms Generate Infinite Byte$</h2>
<h3>Description:</h3> Mob Farms can be used to make infinite Byte$.
<h3>Minimal Working Environment Creation:</h3>
<ol>
       <li>Find a mob spawner.</li>
       <li>Dig a 7 x 7 x 7 cube out.</li>
       <li>Continue the list.</li>
</ol>
<h3>Minimal Working Environment Location: 420, 69, 690</h3>
<h3>Potential Solutions: </h3>
<ul>
       <li>Make it so that Byte$ will not drop within 5 blocks of each other.</li>
       <li>Continue the list.</li>
</ul>
<h3>Media: </h3> (Screenshots or video of the problem are always helpful)

**Comment by dr-hextanium** [comment]
<h2>Bypass Protection Radius</h2>
<h3>Description:</h3> It's possible to bypass the protection radius on mob farms.
<h3>Minimal Working Environment Creation:</h3>
<ol>
       <li>Create a classical mob spawner.</li>
       <li>Create a random redstone pulser.</li>
       <li>Create a lot of minecart tracks with many turns, all at least 5 blocks apart</li>
       <li>Hook up the rails to the random pulser.</li>
       <li>Make any mobs that come out of the spawner fall into a minecart, and use a tripwire to activate the scheme.</li>
</ol>
<strong>Note:</strong> This can be done much easier with minimal redstone with a boat.
<h3>Minimal Working Environment Location: None</h3>
<h3>Potential Solutions: </h3>
<ul>
       <li>Do not allow mobs riding in a minecart or boat to drop items.</li>
</ul>
<h3>Media: </h3> (Screenshots or video of the problem are always helpful)

**Comment by pyxld-kris** [comment]
@TechnoDrive Regarding "Bypass Protection Radius"

Mobs only drop Byte$ when killed by players, which should block auto farming. This may change in the future though if we see that it has a negative impact on some sorts of battle mechanics ie. hitting exploding creepers into mobs, fire damage, etc

**Comment by dr-hextanium** [comment]
> @TechnoDrive Regarding "Bypass Protection Radius"
> 
> Mobs only drop Byte$ when killed by players, which should block auto farming. This may change in the future though if we see that it has a negative impact on some sorts of battle mechanics ie. hitting exploding creepers into mobs, fire damage, etc

Great!

**Comment by SticksDev** [comment]
> @TechnoDrive Regarding "Bypass Protection Radius"
> 
> Mobs only drop Byte$ when killed by players, which should block auto farming. This may change in the future though if we see that it has a negative impact on some sorts of battle mechanics ie. hitting exploding creepers into mobs, fire damage, etc

@pyxld-kris Are you checking the entity type on the kill event?

**Comment by pyxld-kris** [comment]
@Thatcooldevguy Yep, every entity has a different set of rules based on their type

You can see the rules here: https://github.com/dev-launchers/minecraft__byte-economy/blob/3348a4a1f444768f6317720182f2ba24e21e2f90/src/main/java/devlaunchers/byteeconomy/ByteEconomy.java#L68
